### PR TITLE
VOXELFORMAT: Ability to scale converted mesh in x,y,z with different values.

### PR DESCRIPTION
**Pull request by trsh**

### CODE REVIEW

File: src/modules/voxelformat/MeshExporter.cpp [hunk 1]
```diff
@@ -30,7 +30,17 @@ bool MeshExporter::saveGroups(const SceneGraph& sceneGraph, const core::String &
 	const bool mergeQuads = core::Var::get(cfg::VoxformatMergequads, "true", core::CV_NOPERSIST)->boolVal();
 	const bool reuseVertices = core::Var::get(cfg::VoxformatReusevertices, "true", core::CV_NOPERSIST)->boolVal();
 	const bool ambientOcclusion = core::Var::get(cfg::VoxformatAmbientocclusion, "false", core::CV_NOPERSIST)->boolVal();
+
 	const float scale = core::Var::get(cfg::VoxformatScale, "1.0", core::CV_NOPERSIST)->floatVal();
+
+	float scaleX = core::Var::get(cfg::VoxformatScaleX, "1.0", core::CV_NOPERSIST)->floatVal();
```
Comment (mgerhardy) [inline src/modules/voxelformat/MeshExporter.cpp:36]: Here `core::Var::getSafe` should be used (unlike the others here - they are still doing it wrong, too) - the cvars should be created with `core::Var::get` in the `construct()` methods. See e.g. the places where e.g. `cfg::VoxformatMergequads` is created - (`VoxConvert::construct()` and `SceneManager::construct()`)

this allows use to show the cvars in `./vengi-voxconvert --help`

The documentation should get updated to list those new cvars, too (see Formats.md)

**Review by mgerhardy — COMMENTED** [review]
The commit should get a new commit message - I do add prefixes for the modules - in this case `VOXELFORMAT: message` - see the existing commits.

**Comment by mgerhardy** [comment]
The man page also need updates: voxconvert.man.in

Maybe the `voxformat_scale` cvar is no longer needed now?

File: src/modules/voxelformat/MeshExporter.h [hunk 2]
```diff
@@ -23,7 +23,7 @@ class MeshExporter : public Format {
 		glm::vec3 size {0.0f};
 	};
 	using Meshes = core::DynamicArray<MeshExt>;
-	virtual bool saveMeshes(const Meshes& meshes, const core::String &filename, io::SeekableWriteStream& stream, float scale = 1.0f, bool quad = false, bool withColor = true, bool withTexCoords = true) = 0;
+	virtual bool saveMeshes(const Meshes& meshes, const core::String &filename, io::SeekableWriteStream& stream, glm::vec3 scale = glm::vec3(1.0f), bool quad = false, bool withColor = true, bool withTexCoords = true) = 0;
```
[the remark this answers is not in the document]
Comment (trsh) [reply src/modules/voxelformat/MeshExporter.h:26]: OK

File: src/modules/voxelformat/MeshExporter.cpp (hunk 1 above)
Comment (trsh) [reply src/modules/voxelformat/MeshExporter.cpp:36]: OK

**Comment by trsh** [comment]
> The man page also need updates: voxconvert.man.in
> 
> Maybe the `voxformat_scale` cvar is no longer needed now?

I think we should keep it. Shorthand for scaling all at once.

**Pull request closed without merging**